packages/relay/src/lib/eth.ts

```diff
@@ -481,36 +483,49 @@ export class EthImpl implements Eth {
     this.logger.trace('sendRawTransaction(transaction=%s)', transaction);
+
+    const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
+
     try {
       // Convert from 0xabc format into a raw Uint8Array of bytes and execute the transaction
-      const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
       const contractExecuteResponse = await this.sdkClient.submitEthereumTransaction(transactionBuffer);
 
-      // Wait for the record from the execution.
-      const record = await this.sdkClient.getRecord(contractExecuteResponse);
-      if (record.ethereumHash == null) {
-        throw new Error('The ethereumHash can never be null for an ethereum transaction, and yet it was!!');
-      }
-      const txHash = EthImpl.prepend0x(Buffer.from(record.ethereumHash).toString('hex'));
-
-      // If the transaction succeeded, create a new block for the transaction.
-      const mostRecentBlock = await this.mirrorNode.getMostRecentBlock();
-      this.logger.debug('mostRecentBlock=%o', mostRecentBlock);
-      let block = mostRecentBlock;
-      if (record.receipt.status == Status.Success) {
-        block = new CachedBlock(mostRecentBlock, txHash);
-        this.mirrorNode.storeBlock(block);
-      }
+      try {
+        // Wait for the record from the execution.
+        const record = await this.sdkClient.getRecord(contractExecuteResponse);
+        if (record.ethereumHash == null) {
+          throw new Error('The ethereumHash can never be null for an ethereum transaction, and yet it was!!');
+        }
+        const txHash = EthImpl.prepend0x(Buffer.from(record.ethereumHash).toString('hex'));
+
+        // If the transaction succeeded, create a new block for the transaction.
+        const mostRecentBlock = await this.mirrorNode.getMostRecentBlock();
+        this.logger.debug('mostRecentBlock=%o', mostRecentBlock);
+        let block = mostRecentBlock;
+        if (record.receipt.status == Status.Success) {
+          block = new CachedBlock(mostRecentBlock, txHash);
+          this.mirrorNode.storeBlock(block);
+        }
 
-      // Create a receipt. Register the receipt in the cache and return the tx hash
-      if (block == null) {
-        this.logger.error('Failed to get a block for transaction');
-        return '';
+        // Create a receipt. Register the receipt in the cache and return the tx hash
+        if (block == null) {
+          this.logger.error('Failed to get a block for transaction');
+          return '';
+        }
+        const computedHash = EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
+
+		if (computedHash != txHash) {
+			this.logger.error("EthereumHash didn't match computed hash");
+		}
+        return txHash;
+      } catch (e) {
+        this.logger.error(e,
+            'Failed to handle sendRawTransaction cleanly for transaction %s, returning computed hash', transaction);
```

Comment:
Need to update error message. There's more specific logic here which includes the record retrieval and a mirror node call and hash calculation.
So it should differentiate from the general error handle at the end of the method

Reply:
This error is for a failed handle, the outer error is for a failed submission. What would you like for this to contain?

Reply:
The inner one implies a non-precheck error.  The comment should include something alluding to that like "failed at submit" and this one is "failed at receipt retrieval"